docker-apps/rest-proxy/src/server.ts

```diff
@@ -0,0 +1,52 @@
+import { createRestManager, type RequestMethods } from '@discordeno/rest'
+// import { AUTHORIZATION_TOKEN, DISCORD_TOKEN, PORT } from './config.js'
+import { buildFastifyApp } from './fastify.js'
+
+const app = buildFastifyApp()
+
+if (!app.config.DISCORD_TOKEN || !app.config.AUTHORIZATION_TOKEN) {
+  console.error('Missing environment variables. Both DISCORD_TOKEN and AUTHORIZATION_TOKEN are required.')
+  process.exit(1)
+}
+
+const discordRestManager = createRestManager({
+  token: app.config.DISCORD_TOKEN,
+})
+
+app.addHook('onRequest', async (request, reply) => {
+  if (request.headers.authorization !== app.config.AUTHORIZATION_TOKEN) {
+    reply.status(401).send({
+      message: 'Credentials not valid.',
+    })
+  }
+})
+
+app.all('/*', async (request, reply) => {
+  const url = request.originalUrl
+
+  try {
+    const result = await discordRestManager.makeRequest(request.method as RequestMethods, url, {
```

Comment:
In the [current rest proxy](https://github.com/discordeno/discordeno/blob/d81b28a24bb449ebc0aac169cfb53ce098f6986c/proxies/rest/src/index.ts#L24) we have a section that determines if a version is passed through. will this still be required for the proxy to function?

Reply:
Tested this and it doesn't seem like it matters.